src/librbd/ManagedLock.cc

```diff
@@ -750,6 +751,11 @@ void ManagedLock<I>::handle_shutdown_pre_release(int r) {
     m_state = STATE_SHUTTING_DOWN;
   }
 
+  if (r < 0) {
+    post_release_lock_handler(true, r, create_context_callback<
+            ManagedLock<I>, &ManagedLock<I>::handle_shutdown_post_release>(this));
+    return;
+  }
```

Comment:
is it ok to skip calling rados::cls::lock::unlock(&op, RBD_LOCK_NAME, m_cookie);
if possible to simply revise unblock_writes() as follows:
if (!m_write_blockers)
  return

Reply:
Updated to avoid skipping any steps. Just record the error result rightly.